cx/telegram/server.js

```diff
@@ -31,64 +32,78 @@ const {SessionsClient} = require('@google-cloud/dialogflow-cx');
  */
 const client = new SessionsClient({apiEndpoint: locationId + '-dialogflow.googleapis.com'});
 
-/**
- * Converts Telgram request to a detectIntent request.
- */
+// Converts Telgram request to a detectIntent request.
 function telegramToDetectIntent(telegramRequest, sessionPath){
-
-  const request = {
-    session: sessionPath,
-    queryInput: {
-      text: {
-        text: telegramRequest.message.text,
-      },
-      languageCode,
-    },
-  };
-
-  return request;
-}
-
-/**
- * Converts detectIntent response to a Telegram text message request. 
- */
-function detectIntentToTelegramMessage(response,chatId){
-  agentResponse = '';
-    
-  for (const message of response.queryResult.responseMessages) {
-    if (message.text) {
-      agentResponse += `${message.text.text}\n`;
-    };
-  };
-    
-  if(agentResponse.length != ''){
     const request = {
-      chat_id: chatId,
-      text: agentResponse
+        session: sessionPath,
+        queryInput: {
+            text: {
+                text: telegramRequest.message.text,
+            },
+            languageCode,
+        }
     };
+
     return request;
-  };
-};
+}
+
+// Converts detectIntent responses to Telegram message requests.
+async function convertToTelegramMessage(responses, chatId) {
+    let replies = [];
+
+    for(let response of responses.queryResult.responseMessages) {
+        let reply;
+
+        switch(true) {
+            case response.hasOwnProperty('text'): {
+                reply = {
+                    chat_id: chatId,
+                    text: response.text.text.join()
+                };  
+            }
+            break;
```

Comment:
I think it's better to put break; into {...} block

Reply:
Restructured the code.